### PR TITLE
[Develop] Mauricio - Tests board

### LINKED ISSUE #17: Tests - Boards

### PR DESCRIPTION
**Pull request by Snip-77**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,7 +1,7 @@
 filterTags=
 requestLibrary=restassured
 baseUri=https://api.trello.com/
-apiKey=
-apiToken=
+apiKey=f75693d362464a5ef38a3a3a1265b6b4
+apiToken=d623943d90a0049dc8b54affd1034e51ee4496a5f942f492fb613ed3fde5f781
```
Comment (nanrehd) [inline gradle.properties]: Please remove these credentials

File: src/test/resources/schemas/boards/putBoardSchema.json [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    "closed",
+    "idOrganization",
+    "idEnterprise",
+    "pinned",
+    "url",
+    "shortUrl",
+    "prefs",
+    "labelNames"
+  ]
+}
```
Comment (marcos-x86) [inline src/test/resources/schemas/boards/putBoardSchema.json]: Please consider adding an empty line at the end of the file.

File: src/test/java/org/fundacionjala/trello/hooks/CommonHooks.java [hunk 3]
```diff
@@ -33,7 +32,6 @@ public CommonHooks(final Context context) {
      */
     @Before(order = BUILD_REQ_SEC_ORDER_VALUE)
     public void buildDefaultReqSpec() {
-        RequestManagerRestAssured.displayFiltersData(); // Only for help
```
Comment (marcos-x86) [inline src/test/java/org/fundacionjala/trello/hooks/CommonHooks.java]: Please consider not deleting this line, you can add `//` to comment this statement.

File: src/test/resources/features/boards/DELETE_Board.feature [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    Then I validate the response has status code 200
+
+  @negative
+  Scenario Outline: Delete a board with an empty id
+    When I send a DELETE request to "/boards/<value>"
+    Then I validate the response has status code 404
+    Examples:
+      | value   |
+      | {empty} |
+      |         |
```
Comment (marcos-x86) [inline src/test/resources/features/boards/DELETE_Board.feature]: This case is not necessary, please delete this data entry.

File: src/test/resources/features/boards/DELETE_Board.feature [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      | 5ebcf20b2a5ab832271a56fA |
+
+  @negative
+  Scenario Outline: Delete a board with an invalid id
+    When I send a DELETE request to "/boards/<value>"
+    Then I validate the response has status code 400
+    Examples:
+      | value                         |
+      | gggggggggggggggggggggggg      |
+      | GGGGGGGGGGGGGGGGGGGGGGGG      |
```
Comment (marcos-x86) [inline src/test/resources/features/boards/DELETE_Board.feature]: This case is not necessary, it is the same as the previous line, please delete this data entry.

File: src/test/resources/features/boards/DELETE_Board.feature [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      | $$$$$$$$$$$$$$$$$$$$$$$$      |
+      | aaaaaaaaaaaaaaaaaaa           |
+      | aaaaaaaaaaaaaaaaaaaaaaaaaaaaa |
```
Comment (marcos-x86) [inline src/test/resources/features/boards/DELETE_Board.feature]: Please consider just **one** of these cases: `aaaaaaaaaaaaaaaaaaa` or `aaaaaaaaaaaaaaaaaaaaaaaaaaaaa`.

File: src/test/resources/features/boards/GET_Board.feature [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      | prefs.background | {board.prefs.background} |
+
+  @negative
+  Scenario Outline: Try to get a board with an invalid id
+    When I send a GET request to "/boards/<value>"
+    Then I validate the response has status code 400
+    Examples:
+      | value                         |
+      | gggggggggggggggggggggggg      |
+      | GGGGGGGGGGGGGGGGGGGGGGGG      |
```
Comment (marcos-x86) [inline src/test/resources/features/boards/GET_Board.feature]: This case is not necessary, it is the same as the previous line, please delete this data entry.

File: src/test/resources/features/boards/GET_Board.feature [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+      | $$$$$$$$$$$$$$$$$$$$$$$$      |
+      | aaaaaaaaaaaaaaaaaaa           |
+      | aaaaaaaaaaaaaaaaaaaaaaaaaaaaa |
```
Comment (marcos-x86) [inline src/test/resources/features/boards/GET_Board.feature]: Please consider just **one** of these cases: `aaaaaaaaaaaaaaaaaaa` or `aaaaaaaaaaaaaaaaaaaaaaaaaaaaa`.

File: src/test/resources/features/boards/GET_Board.feature [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+  @negative
+  Scenario Outline: Try to get a board with empty ids
+    When I send a GET request to "/boards/<value>"
+    Then I validate the response has status code 404
+    Examples:
+      | value   |
+      | {empty} |
+      |         |
```
Comment (marcos-x86) [inline src/test/resources/features/boards/GET_Board.feature]: This case is not necessary, please remove it.

File: src/test/resources/features/boards/POST_BoardNew.feature [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+      | name | <value> |
+    And I save the id value to clean "board" workspace
+    Then I validate the response has status code 200
+    And I validate the response body should match with "boards/boardSchema.json" JSON schema
+    And I validate the response contains the following data
+      | name | <name> |
+    Examples:
+      | value                     | name                      |
+      | 5ebcf20b25ab832271a56fb%$ | 5ebcf20b25ab832271a56fb%$ |
+      | $$$$$$$$$$$$$$$$$$$$$$$$$ | $$$$$$$$$$$$$$$$$$$$$$$$$ |
```
Comment (marcos-x86) [inline src/test/resources/features/boards/POST_BoardNew.feature]: This case is the same as the previous one, please consider removing it.

File: src/test/resources/features/boards/POST_BoardNewJson.feature [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+      """
+    And I save the id value to clean "board" workspace
+    Then I validate the response has status code 200
+    And I validate the response body should match with "boards/boardSchema.json" JSON schema
+    And I validate the response contains the following data
+      | name | <name> |
+    Examples:
+      | value                     | name                      |
+      | 5ebcf20b25ab832271a56fb%$ | 5ebcf20b25ab832271a56fb%$ |
+      | $$$$$$$$$$$$$$$$$$$$$$$$$ | $$$$$$$$$$$$$$$$$$$$$$$$$ |
```
Comment (marcos-x86) [inline src/test/resources/features/boards/POST_BoardNewJson.feature]: This case is the same as the previous, please consider removing it.

File: src/test/resources/features/boards/PUT_BoardUpdates.feature [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+  Scenario Outline: Update a board with invalid ids
+    When I send a PUT request to "/boards/<value>" with the following parameters
+      | name             | Board name updated           |
+      | desc             | Description has been updated |
+      | prefs/background | lime                         |
+    Then I validate the response has status code 400
+    Examples:
+      | value                         |
+      | gggggggggggggggggggggggg      |
+      | GGGGGGGGGGGGGGGGGGGGGGGG      |
```
Comment (marcos-x86) [inline src/test/resources/features/boards/PUT_BoardUpdates.feature]: This case is the same as the previous, please consider deleting it.

File: src/test/resources/features/boards/PUT_BoardUpdates.feature [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+      | $$$$$$$$$$$$$$$$$$$$$$$$      |
+      | aaaaaaaaaaaaaaaaaaa           |
+      | aaaaaaaaaaaaaaaaaaaaaaaaaaaaa |
```
Comment (marcos-x86) [inline src/test/resources/features/boards/PUT_BoardUpdates.feature]: Please consider only **one** of these cases: `aaaaaaaaaaaaaaaaaaa` or `aaaaaaaaaaaaaaaaaaaaaaaaaaaaa`

File: src/test/resources/features/boards/PUT_BoardUpdates.feature [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+  Scenario Outline: Update a board with empty ids
+    When I send a PUT request to "/boards/<value>" with the following parameters
+      | name             | Board name updated           |
+      | desc             | Description has been updated |
+      | prefs/background | lime                         |
+    Then I validate the response has status code 404
+    Examples:
+      | value   |
+      | {empty} |
+      |         |
```
Comment (marcos-x86) [inline src/test/resources/features/boards/PUT_BoardUpdates.feature]: This case is not necessary, please delete it.

File: src/test/resources/features/boards/PUT_BoardUpdatesJson.feature [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+	    "name" : "Board name updated",
+	    "desc" : "Description has been updated using a Json",
+	    "prefs/background" : "lime"
+      }
+      """
+    Then I validate the response has status code 400
+    Examples:
+      | value                         |
+      | gggggggggggggggggggggggg      |
+      | GGGGGGGGGGGGGGGGGGGGGGGG      |
```
Comment (marcos-x86) [inline src/test/resources/features/boards/PUT_BoardUpdatesJson.feature]: This case is the same as the previous, please consider deleting it.

File: src/test/resources/features/boards/PUT_BoardUpdatesJson.feature [hunk 16]
```diff
@@ -0,0 +1,132 @@
...
+      | $$$$$$$$$$$$$$$$$$$$$$$$      |
+      | aaaaaaaaaaaaaaaaaaa           |
+      | aaaaaaaaaaaaaaaaaaaaaaaaaaaaa |
```
Comment (marcos-x86) [inline src/test/resources/features/boards/PUT_BoardUpdatesJson.feature]: Please consider only **one** of these cases: `aaaaaaaaaaaaaaaaaaa` or `aaaaaaaaaaaaaaaaaaaaaaaaaaaaa`

File: src/test/resources/features/boards/PUT_BoardUpdatesJson.feature [hunk 17]
```diff
@@ -0,0 +1,132 @@
...
+	    "name" : "Board name updated",
+	    "desc" : "Description has been updated using a Json",
+	    "prefs/background" : "lime"
+      }
+      """
+    Then I validate the response has status code 404
+    Examples:
+      | value   |
+      | {empty} |
+      |         |
```
Comment (marcos-x86) [inline src/test/resources/features/boards/PUT_BoardUpdatesJson.feature]: This case is not necessary, please delete it.

**Review by marcos-x86 — CHANGES_REQUESTED** [review]

**Review by marcos-x86 — APPROVED** [review]

**Pull request merged**